server/src/main/java/org/sonarsource/ldap/server/ApacheDS.java

```diff
@@ -74,16 +75,20 @@ private ApacheDS(String realm, String baseDn) {
     ldapServer = new LdapServer();
   }
 
-  public static ApacheDS start(String realm, String baseDn, String workDir) throws Exception {
+  public static ApacheDS start(String realm, String baseDn, String workDir, Integer port) throws Exception {
     return new ApacheDS(realm, baseDn)
       .startDirectoryService(workDir)
       .startKdcServer()
-      .startLdapServer()
+      .startLdapServer(port == null ? AvailablePortFinder.getNextAvailable(1024) : port)
       .activateNis();
   }
 
+  public static ApacheDS start(String realm, String baseDn, String workDir) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "workDir". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)